includes/class-wc-payments-order-service.php

```diff
@@ -1438,32 +1438,25 @@ private function update_order_status( $order, $order_status, $intent_id = '' ) {
 	 * Takes an intent object or array and returns our needed data as an array.
 	 * This is needed due to intents can either be objects or arrays.
 	 *
-	 * @param object|array $intent The intent to pull the data from.
+	 * @param WC_Payments_API_Abstract_Intention $intent  Setup or payment intent to pull the data from.
 	 *
 	 * @return array The data we need to continue processing.
 	 */
 	private function get_intent_data( $intent ): array {
-		$intent_data = [];
-		if ( is_array( $intent ) ) {
-			$intent_data = [
-				'intent_id'           => $intent['id'],
-				'intent_status'       => $intent['status'],
-				'charge_id'           => $intent['charge_id'] ?? '',
-				'fraud_outcome'       => $intent['fraud_outcome'] ?? '',
-				'payment_method_type' => $intent['payment_method_type'] ?? '',
-			];
-		} elseif ( is_object( $intent ) ) {
-			$charge               = $intent->get_charge();
-			$payment_method_types = $intent->get_payment_method_types();
-
-			$intent_data = [
-				'intent_id'           => $intent->get_id(),
-				'intent_status'       => $intent->get_status(),
-				'charge_id'           => $charge ? $charge->get_id() : null,
-				'fraud_outcome'       => $intent->get_metadata()['fraud_outcome'] ?? '',
-				'payment_method_type' => 1 === count( $payment_method_types ) ? $payment_method_types[0] : '',
-			];
+
+		$intent_data = [
+			'intent_id'           => $intent->get_id(),
+			'intent_status'       => $intent->get_status(),
+			'charge_id'           => '',
+			'fraud_outcome'       => $intent->get_metadata()['fraud_outcome'] ?? '',
+			'payment_method_type' => $intent->get_payment_method_type(),
+		];
+
+		if ( is_a( $intent, WC_Payments_API_Payment_Intention::class ) ) {
```

Comment:
```suggestion
		if ( $intent instanceof WC_Payments_API_Payment_Intention ) ) {
```

Let's use `instanceof` instead of `is_a()`. It's functionally the same, but VS Code understands it a bit better. 
<img width="936" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5311119/c653c684-29cb-45fd-99fb-b2849a8b5f75">
<img width="849" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5311119/af753c45-f344-436b-a4ae-e16b03d8556a">

Reply:
PHPStorm seems smarter :D 

![Markup on 2023-07-14 at 14:21:39](https://github.com/Automattic/woocommerce-payments/assets/10045087/6b4ca50a-37aa-4c4b-bd7b-96452b407a1a)

Anyway, I've made the change: c9c09c1461876f51f81a4a54925b11b9455f47a2